geps/gep-3440/index.md

```diff
@@ -0,0 +1,281 @@
+# GEP-3440: Gateway API Support for gRPC Retries
+
+* Issue: [#3440](https://github.com/kubernetes-sigs/gateway-api/issues/3440)
+* Status: Provisional
+
+## TLDR
+This proposal introduces support for gRPC retries in the Gateway API,
+allowing for configuration of retry attempts, backoff duration, and retryable status codes for gRPC routes.
+
+## Goals
+
+- To allow specification of gRPC status codes that should be retried.
+- To allow specification of the maximum number of times to retry a gRPC request.
+- To allow specification of the minimum backoff interval between retry attempts for gRPC requests.
+- Retry configuration must be applicable to most known Gateway API implementations for gRPC.
+- To define any interaction with configured gRPC timeouts and backoff.
+
+## Non-Goals
+
+- No standard APIs for advanced retry logic, such as integrating with rate-limiting headers.
+- No default retry policies for all routes within a namespace or for routes tied to a specific Gateway.
+- No support for detailed backoff adjustments, like fine-tuning intervals, adding jitter, or setting max duration caps.
+- No retry support for streaming or bidirectional APIs (maybe considered in future proposals).
```

Comment:
How is this enforced in the API specification?

Reply:
Thanks for calling that out. The API doesn't have a way to enforce this non-goal. 

I am thinking of three ways to do that:
1. Adding a restriction in the API documentation clarifying that retries apply only to unary calls, with a potential future option to expand to streaming. Something among the line as:
```
// Note: **Retries are supported only for unary gRPC calls.**
// Implementations MUST NOT apply retries to streaming or bidirectional
// gRPC calls, as these types of calls are stateful and retrying them
// could result in data loss or duplication.
```
2. Explicit Field: Add a UnaryOnly field (e.g., UnaryOnly bool) that makes it clear retries are restricted to unary calls.
3. Remove this restriction and let users choose whether to apply retries on any gRPC call type.

Would like to hear your thoughts on this.